### DISCUSSION TITLE
Add delete functionality in the Layer Manager.

### DISCUSSION THREAD

**Discussion by Duvan1** [GrapesJS/grapesjs#5048] [Q&A] [answered]
Using the layerManager.onRender() I was able to alter the layer manager dom to look like in the image, I was able to add a delete button, but I would like to add functionality that allows me to delete the selected element.
Reviewing the [source code](https://github.com/GrapesJS/grapesjs/blob/d802eba08559a3e1c7830e937a78b0b2a417f2e4/src/navigator/view/ItemView.js#L12-L22), I see that there is a list of events with their respective function, I would like to know if there is any way to add another functionality that allows to perform the deletion as suggested in [this discussion](https://github.com/GrapesJS/grapesjs/issues/2364#issuecomment-815574702).
Thank you very much in advance, I hope someone can help me with this.
![image](https://user-images.githubusercontent.com/28686863/231786070-9bbf4bd9-05a3-4f7e-800b-43d918119796.png)

**Comment by artf** [comment, marked as answer]
You can try to make use of [`extend`](https://github.com/GrapesJS/grapesjs/blob/976e791c5abd6f876a0f29b58de32ff1549f32c2/src/navigator/config/config.ts#L108) option but I'd highly suggest to rely on the Layers API to create your [own UI for the layer manager](https://grapesjs.com/docs/modules/Layers.html#customization), that will give a full control
